### PR TITLE
[SSDK-399] Add Discover.Options(country, proximity, origin)

### PR DESCRIPTION
**Pull request by aokj4ck**
### Description

**Ticket**: [SSDK-399](https://mapbox.atlassian.net/browse/SSDK-399) 

- Add Country.default: Self? to auto-detect the current device region and provide a value if found
- Add proximity and origin CLLocationCoordinate2D fields to Discover.Option and forward these to SearchOption parameters
- This fixes an issue when using search-along-route to query category results by providing a country, proximity, and origin parameters to Discover.Options.

### Checklist
- [x] Update `CHANGELOG`

[SSDK-399]: https://mapbox.atlassian.net/browse/SSDK-399?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Sources/MapboxSearch/PublicAPI/Common/Models/Country/Country.swift [hunk 1]
```diff
@@ -12,4 +12,23 @@ public struct Country: Equatable {
...
+    }
+
+    /// Detect the system region ISO3166\_1\_Alpha2 identifier and return an instance for it
+    static var `default`: Self? {
+        if #available(iOS 16, *) {
+            return Country(countryCode: Locale.current.region?.identifier ?? "")
+        } else {
+            let regionComponents = Locale.current.identifier.components(separatedBy: "_")
+            if regionComponents.count >= 2 {
+                let countryIdentifier = regionComponents[1]
```
Comment (chizhavko) [inline Sources/MapboxSearch/PublicAPI/Common/Models/Country/Country.swift:27]: Hey @aokj4ck is `countryIdenitifier` only the last component in `regionComponents` or it is just the second item all the time? 

just smal nitpick from my side: is it possible to use `regionComponents?.last` or `regionComponents?.first` to leave all these checks on compiler shoulders? Instead to access item via index in array

**Review by chizhavko — APPROVED** [review]
LGTM

File: Sources/MapboxSearch/PublicAPI/Use Cases/Discover API/Models/Discover+Options.swift [hunk 2]
```diff
@@ -15,12 +15,27 @@ extension Discover {
...
             limit: Int = 10,
-            language: Language? = nil
+            language: Language? = nil,
+            country: Country? = nil,
+            proximity: CLLocationCoordinate2D? = nil,
+            origin: CLLocationCoordinate2D? = nil
         ) {
             self.limit = limit
             self.language = language ?? .default
+            self.country = country ?? .default
```
Comment (kried) [inline Sources/MapboxSearch/PublicAPI/Use Cases/Discover API/Models/Discover+Options.swift:36]: From what I know sending the `country` and `language` parameters could trigger enabling federation search, that is, an additional search data source. It seems like unwanted behavior to always force to set the country parameter.

Comment (aokj4ck) [reply Sources/MapboxSearch/PublicAPI/Use Cases/Discover API/Models/Discover+Options.swift:36]: Oh that's a great point thank you. Yes I think removing the `?? .default` assignment will be good.

File: Sources/MapboxSearch/PublicAPI/Common/Models/Country/Country.swift (hunk 1 above)
Comment (aokj4ck) [reply Sources/MapboxSearch/PublicAPI/Common/Models/Country/Country.swift:27]: Great question, it **is** the second item all the time and there **may be** more items.

For example if you:
1. Add an iOS 15 simulator (oldest simulator available in Xcode 14 and that uses this API)
2. Change the Demo app > Edit Scheme > Run > Options > App Language and App Region
3. Set a breakpoint on this line
4. Build and run
5. Go to Discover tab and tap "Search in this area"

You will find that there are scenarios where the locale has more components such as Language = Chinese (Hong Kong) and Region = Hong Kong will return the locale `"en_HK_HK"`. There are various other examples and accessing the second items in the regionComponents is safest to cover all scenarios.

Screenshot 2024-02-14 at 10 16 17

Screenshot 2024-02-14 at 10 18 44

File: Sources/MapboxSearch/PublicAPI/Use Cases/Discover API/Discover.swift [hunk 3]
```diff
@@ -59,9 +59,11 @@ extension Discover {
         userActivityReporter.reportActivity(forComponent: "discover-search-nearby")
 
         let searchOptions = SearchOptions(
+            countries: [options.country?.countryCode].compactMap { $0 },
             languages: [options.language.languageCode],
             limit: options.limit,
-            proximity: proximity
+            proximity: options.proximity ?? proximity,
```
Comment (kried) [inline Sources/MapboxSearch/PublicAPI/Use Cases/Discover API/Discover.swift:65]: As for me, here it should be rather
```suggestion
            proximity: proximity
```
This method takes non-optional argument, so it should be prioritized over `options.proximity`

**Review by kried — DISMISSED** [review]

File: Sources/MapboxSearch/PublicAPI/Use Cases/Discover API/Discover.swift (hunk 3 above)
Comment (aokj4ck) [reply Sources/MapboxSearch/PublicAPI/Use Cases/Discover API/Discover.swift:65]: Yeah you're right to use the non-optional with priority.

**Review by kried — APPROVED** [review]

**Pull request merged**